### ISSUE TITLE
Should Access plugin allow JWT's without aud, iss, exp, nbf?

### ISSUE THREAD

**Issue by jahands** [cloudflare/pages-plugins#19]
I was copying some of the Access plugin jwt validator code and noticed it allows JWT payloads without aud, iss, exp, nbf fields. Is this intended? Will Access JWTs ever be missing these fields? It seems odd to allow tokens without them.

**Comment by jahands** [comment]
Okay it seems `nbf` is not present on requests using service auth

**Comment by GregBrimble** [comment]
Yeah, you're probably correct. My thinking was that, if the payload didn't contain these fields, at least we knew it had been signed by Cloudflare, so could probably trust it. But maybe we should mandate that some fields are present.

I've filed a docs ticket to get clarity for us and everyone else: https://github.com/cloudflare/cloudflare-docs/issues/7412

**Comment by GregBrimble** [comment]
Also related: https://developers.cloudflare.com/api-shield/security/jwt-validation/configure/#performing-jwt-validation

> Should the JWT contain an EXP claim (expiration time), we validate that the JWT is not expired.

> Should the JWT contain a NBF claim (not before time), we validate that the JWT is already valid.

They seem to do something similar to what I've done here: validating fields if present. `aud` and `iss` perhaps should also be mandated, but I'll wait to see what the ZT team say about it in that docs issue.
